### PR TITLE
feat: support timestamp comparison

### PR DESCRIPTION
**Pull request by maksimtor**
Support timestamp comparison using operators <, <=, >, >=.

### CODE REVIEW

**Comment by bruceg** [comment]
Closes #579

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## unreleased
 * `parse_regex_all` `pattern` param  can now be resolved from a variable
 * fixed `parse_json` data corruption issue for numbers greater or equal to `i64::MAX` 
+* support timestamp comparison using operators <, <=, >, >=
```
Comment (pront) [inline CHANGELOG.md:6]: 💭 Can you add an example here as well: https://vector.dev/docs/reference/vrl/expressions/#comparison-examples

File: lib/tests/tests/expressions/comparison/ge.vrl [hunk 2]
```diff
@@ -21,6 +21,10 @@
     "foo" >= "foo",
     "bar" >= "foo",
 
+    t'2024-04-04T22:22:22.234142Z' >= t'2024-05-04T22:22:22.234142Z',
+    t'2024-04-04T22:22:22.234142Z' >= t'2024-04-04T22:22:22.234142Z',
+    t'2024-05-04T22:22:22.234142Z' >= t'2024-04-04T22:22:22.234142Z',
```
Comment (pront) [inline lib/tests/tests/expressions/comparison/ge.vrl:26]: let's add a few examples with different timezones e.g.
```
t'2024-05-04T22:22:22.234142+01:00' > t'2024-05-04T22:22:22.234142+01:00',
```

**Review by pront — COMMENTED** [review]
Thank you @maksimtor for this PR!

File: CHANGELOG.md (hunk 1 above)
Comment (maksimtor) [reply CHANGELOG.md:6]: Sure, will try to remember to do it a bit later.

**Review by pront — APPROVED** [review]
Thank you!

**Pull request merged**